hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -166,12 +166,12 @@ public static OffsetRange[] computeOffsetRanges(Map<TopicPartition, Long> fromOf
           if (toOffset == range.untilOffset()) {
             exhaustedPartitions.add(range.partition());
           }
-          allocedEvents += toOffset - range.fromOffset();
           // We need recompute toOffset if allocedEvents larger than actualNumEvents.
```

Comment:
maybe "We need recompute toOffset if we have allocated more than actualNumEvents. total allocated events = allocedEvents + (toOffset - range.fromOffset())"